### ISSUE TITLE
updating tag values in opentsdb

### ISSUE THREAD

**Issue by gsaray101** [OpenTSDB/opentsdb#486]
Hi group, I really need help updating tag values in opentsdb. I read through the docs but still not clear the best way to do this.

I am pushing cpu, memory utilization data to opentsdb. I would like to tag each server with app name so that I can organize my data based on application, server etc. There are instances of servers where I cannot classified for an application there I will tag the this metrics as other for example

pctCPU <time> 20 server=j2kdc1  app=Other

At some point if I know the name of the application this server j2kdc1 belongs to, I want to be able update the tag from Other to some application name like Messaging.

pctCPU <time> 20 server=j2kdc1 app=Messaging

Does any body have any idea, how could I easily do this? Any thoughts?

**Comment by michaelbironneau** [comment]
I think if "at some point" means "in the next hour" then you can just re-insert the same point with the different tag and it should work. From the docs

>Writing data points in OpenTSDB is generally idempotent within an hour of the original write. This >means you can write the value 42 at timestamp 1356998400 and then write 42 again for the same >time and nothing bad will happen

If it's longer than one hour I think there will be a way to do this in 2.1 by setting ` tsd.storage.fix_duplicates` to `true`, which deduplicates rows according to last-write-wins, so inserts will be idempotent.

**Comment by manolama** [comment]
Unfortunately don't have a way to do that right now. It's pretty involved in that you need to scan all of the rows in HBase for the given metric, write new rows with the new tags, then delete the old ones. I may get to it eventually but if you have time to work on it, that'd be great!

**Comment by gsaray101** [comment]
@manolama OpenTSDB is great idea for performance and storage but for it to be really enterprise ready, we have to be able to update the tags on the fly since tags are the integral part of this technology. If you have 10s of thousands of servers etc, writing rows with new tags and deleting the old ones would take very long time to complete and in the mean time stopping updating. That will not work. Is there any other way you could think of to address is this in the mean time?

**Comment by gsaray101** [comment]
@manolama Also, you tagged this as medium but if you really use trees based on tags, I think this feature is extremely important to me.

**Comment by biswaKL** [comment]
Yes this is an very important feature, in data validation point of view, we should be able to tag the data as valid or invalid whenever we want. I am facing a lot of issue because of this and unable to update the tags.
